c10/core/WrapDimMinimal.h

```diff
@@ -1,25 +1,45 @@
 #pragma once
 
+#include <c10/core/SymInt.h>
 #include <c10/util/Exception.h>
 
 namespace c10 {
 
 namespace detail {
-C10_API int64_t
-maybe_wrap_dim_slow(int64_t dim, int64_t dim_post_expr, bool wrap_scalar);
-}
+template <typename T>
+C10_API T maybe_wrap_dim_slow(T dim, T dim_post_expr, bool wrap_scalar);
```

Comment:
declaring a template but not giving any implementations of it doesn't do anything

Reply:
oh you're playing tricks with defining the template instantiations in the cpp file lol

Reply:
idk if this is idiomatic or not, but I certainly learned a new C++ feature today!